EmVer/src/EmVerUtility.cpp

```diff
@@ -0,0 +1,53 @@
+#include <string.h>
+
+#include "mbed.h"
+
+#include "EmVerUtility.hpp"
+
+bool assertPin(PinName pinName, bool expect, int timeout) { // timeout in ms
+    Timer t;
+    DigitalIn pin(pinName);
+    t.start();
+    while(t.read_ms() < timeout) {
+        if(pin == expect) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool assertPinToggle(PinName pinName, unsigned int expectOccurences, int timeout) { // timeout in ms
+    Timer t;
+    DigitalIn pin(pinName);
+    t.start();
+    bool prevStatus = pin;
+    unsigned int occurences = 0;
+    while(t.read_ms() < timeout) {
+        if(pin != prevStatus) {
+            prevStatus = pin;
+            occurences++;
+        }
+    }
+    return occurences == expectOccurences;
+}
+
+bool assertCan(PinName p1, PinName p2, unsigned int expectId, const int* expectData, const int expectLen, int timeout) { // timeout in ms
+    Timer t;
+    CANMessage msg;
+    CAN can(p1, p2);
```

Comment:
This would require testing but I have a feeling this might not work. The reason is that when you create a CAN it does initialization of the pin so if you are asserting lots of different CAN messages it could be a problem. What I would do instead is instead of passing in PinNames just pass in the CAN reference. (Initialize them once at the beginning of the test, there are only 2 possible ones).